taskcluster/script.py

```diff
@@ -151,6 +152,42 @@ def main():
     except subprocess.CalledProcessError as e:
         print('{} attempting netstat'.format(e))
 
+    # enable charging on device if it is disabled
+    #   see https://bugzilla.mozilla.org/show_bug.cgi?id=1565324
+    charging_disabled = False
+    p2_path = "/sys/class/power_supply/battery/input_suspend"
+    g5_path = "/sys/class/power_supply/battery/charging_enabled"
+    p2_batt_input_suspended = bool(int(device.shell_output("cat %s" % p2_path)))
+    g5_charging_enabled = bool(int(device.shell_output("cat %s" % g5_path)))
+    if p2_batt_input_suspended or not g5_charging_enabled:
+        charging_disabled = True
+    if charging_disabled:
+        try:
+            timeout = 10
+            device_name = device.shell_output(
+                "getprop ro.product.model", timeout=timeout
+            )
+            print("Enabling charging...")
+            if device_name == "Pixel 2":
+                device.shell_bool(
+                    "echo %s > %s" % (0, p2_path), root=True, timeout=timeout
+                )
+            elif device_name == "Moto G (5)":
+                device.shell_bool(
+                    "echo %s > %s" % (1, g5_path), root=True, timeout=timeout
+                )
+            else:
+                print(
+                    "WARNING: Unknown device! Not sure how to enable charging for device of type '%s'."
```

Comment:
Hmm, I guess this should be fatal also?

Reply:
Making this fatal will make testing new device more difficult (we'd have to create a new image first), but would remind us to add it to this code.

Reply:
Changed to fatal.